single-campaigns.php

```diff
@@ -1,46 +1,106 @@
 <?php
 /**
- * The Template for displaying all single campaign posts
+ * Template Variables for Campaigns.
  *
- * Methods for TimberHelper can be found in the /lib sub-directory
- *
- * @package  WordPress
- * @subpackage  Timber
- * @since    Timber 0.1
+ * @package P4MT
  */
 
 use Timber\Timber;
 
 // Initializing variables.
 $context = Timber::get_context();
 
-// @var P4_Post object $post
+/**
+ * Post object.
+ *
+ * @var P4_Post $post
+ * */
 $post            = Timber::query_post( false, 'P4_Post' );
 $context['post'] = $post;
 
+// Save custom style settings.
+$custom_styles = [];
+
+$custom_styles['css']['nav_color']     = $post->campaign_nav_color ? ".navbar { background-color: {$post->campaign_nav_color} !important;}" : null;
+$custom_styles['nav_type']             = $post->campaign_nav_type;
+$custom_styles['logo_color']           = $post->campaign_logo_color;
+$custom_styles['css']['header_bg']     = $post->campaign_header_color ? ".page-header { background: {$post->campaign_header_color} !important;}" : null;
+$custom_styles['css']['header_serif']  = $post->campaign_header_serif ? ".page-header { font-family: {$post->campaign_header_serif}!important;}" : null;
+$custom_styles['css']['header_sans']   = $post->campaign_header_sans ? ".page-header { font-family: {$post->campaign_header_sans} !important;}" : null;
+$custom_styles['css']['body_font']     = $post->campaign_body_font ? "body { font-family: '{$post->campaign_body_font}' !important;}" : null;
+$custom_styles['css']['btn_primary']   = $post->campaign_primary_color ? ".btn-primary { background: {$post->campaign_primary_color} !important; border-color: {$post->campaign_primary_color} }" : null;
+$custom_styles['css']['btn_secondary'] = $post->campaign_secondary_color ? ".btn-secondary { background: {$post->campaign_secondary_color} !important; border-color: {$post->campaign_primary_color} }" : null;
```

Comment:
Shouldn't these style live inside a scss file instead?

Reply:
Not sure if that's the way to go? These are meant to be individual overrides of each campaign theme's default styles. So the user can pick a different font or different nav colour from the post's edit page. Or what do you think?

Reply:
I'm thinking about pre-builting the custom css file(s) (naming it per postid/postname) for each specific post/campaign  in a hook like 'save_post' and then enqueing it here. What do you think?

Reply:
@kirdia Or I can make inline-classes for the options? It should be fine for the fonts but things like colors will be a lot of classes for the colours. Is that okay?

Reply:
Actually @comzeradd  is probably right. I'll add these as variables to the SCSS. Would you prefer I make a new file or add to the existing ones?

Reply:
@onyekaa I did not get this. How would you pass the values to the css variables? Shouldn't these be customizable per campaign page?

Reply:
Sorry mistake. I was suggesting making classes for the settings, so for instance, header serif will have a class of `.header-serif`, colors can uses SASS maps for all the color options like `.bg-green-light`. But that feels excessive. 

All the settings currently generate about 5 classes which I just put in a style tag in the head. For example:
`<style>
.navbar { background-color: #8cbf4f !important;}
.page-header { background: #ffffff !important;}
body { font-family: 'lora' !important;}
.btn-primary { background: #fcd00a !important; border-color: #fcd00a }
.btn-secondary { background: #ffffff !important; border-color: #fcd00a }		
</style>`